dask/array/reductions.py

```diff
@@ -181,9 +181,17 @@ def nanprod(a, axis=None, dtype=None, keepdims=False, split_every=None):
             dt = np.empty((1,), dtype=a._dtype).nanprod().dtype
         else:
             dt = None
-        return reduction(a, chunk.nanprod, chunk.prod, axis=axis,
+        return reduction(a, chunk.nanprod, chunk.nanprod, axis=axis,
```

Comment:
Good call, thank you for catching me on this really stupid error!  The `nanprod` always returns at least a 1, likewise for `nansum` returning a 0.  Sorry about the confusion.  I've fixed this.